crates/commonware-node/src/dkg/manager/actor/mod.rs

```diff
@@ -160,444 +137,353 @@ where
-            info!(
-                "no epoch state found on disk - reading validators and public \
-                polynomial from genesis block",
-            );
+        storage: &mut state::Storage<TStorageContext>,
+        mux: &mut MuxHandle<TSender, TReceiver>,
+    ) -> eyre::Result<State>
+    where
+        TStorageContext: commonware_runtime::Metrics + commonware_runtime::Storage,
+        TSender: Sender<PublicKey = PublicKey>,
+        TReceiver: Receiver<PublicKey = PublicKey>,
+    {
+        let state = storage.current();
 
-            let initial_dkg_outcome = PublicOutcome::decode(spec.genesis().extra_data.as_ref())
-                .wrap_err_with(|| {
-                    format!(
-                        "failed decoding the genesis.extra_data field as an \
-                        initial DKG outcome; this field must be set and it \
-                        must be decodable; bytes = {}",
-                        spec.genesis().extra_data.len(),
-                    )
-                })?;
+        self.metrics.reset();
 
-            ensure!(
-                initial_dkg_outcome.epoch.is_zero(),
-                "at genesis, the epoch must be zero, but genesis reported `{}`",
-                initial_dkg_outcome.epoch
-            );
+        self.metrics.dealers.set(state.dealers.len() as i64);
+        self.metrics.players.set(state.players.len() as i64);
+        self.metrics.syncing_players.set(state.syncers.len() as i64);
 
-            let our_share = self.config.initial_share.clone();
-            if let Some(our_share) = our_share.clone() {
-                // XXX: explicitly check the signing key matches the public
-                // polynomial. If it does not, commonware silently demotes the
-                // node to a verifier.
-                //
-                // FIXME: replace this once commonware provides logic to not
-                // degrade the node silently.
-                let signer_or_verifier = Scheme::<_, MinSig>::new(
-                    initial_dkg_outcome.participants.clone(),
-                    &initial_dkg_outcome.public,
-                    our_share,
-                );
-                ensure!(
-                    matches!(signer_or_verifier, Scheme::Signer { .. },),
-                    "incorrect signing share provided: the node would not be a \
-                    signer in the ceremony"
-                );
-            }
+        if let Some(previous) = state.epoch.previous() {
+            // NOTE: State::prune emits an error event.
+            storage.prune(previous).await.wrap_err_with(|| {
+                format!("unable to prune storage before up until epoch `{previous}`",)
+            })?;
+        }
+
+        let all_peers = state.construct_merged_peer_set();
+        self.metrics.peers.set(all_peers.len() as i64);
+        self.config
+            .peer_manager
+            .update(state.epoch.get(), all_peers)
+            .await;
```

Comment:
unsure on the implications right now but we're including inactive peers here as well, if a dealer/player was set to inactive in the contract, maybe we shouldnt connect to it ?